### PR TITLE
Filter dropdown interactions

### PR DESCRIPTION
**Pull request by xtine**
## Summary
- Filter dropdown panel button now shows the loading state
- Shows checked items within dropdown panel
- Ability to remove unchecked dropdown items

## Screenshots
2wmy8n56wy

Requires 18F/openfec-web-app#1395

_cc @noahmanger

### CODE REVIEW

**Comment by xtine** [comment]
Now tag removal will also remove the dropdown checkboxes entirely, and clicking "clear all filters" will move all dropdown checkboxes entirely.

znjsajncer

File: js/dropdowns.js [hunk 1]
```diff
@@ -106,14 +113,69 @@ Dropdown.prototype.handleCheck = function(e) {
   }
 };
 
+Dropdown.prototype.handleDropdownItemClick = function(e) {
+  var $button = $(e.target);
+  var $input = $('#' + $button.data('label'));
```
Comment (noahmanger) [inline js/dropdowns.js]: Let's scope this to `this.$selected`. So: `this.$selected.find('#' + $button.data('label'));`

File: js/dropdowns.js [hunk 2]
```diff
@@ -106,14 +113,69 @@ Dropdown.prototype.handleCheck = function(e) {
...
+
+  if (!$button.hasClass('is-checked')) {
+    $input.click();
+  }
+};
+
+Dropdown.prototype.handleSelectedInputClick = function(e) {
+  var $button = $('button[data-label=' + e.target.id + ']');
```
Comment (noahmanger) [inline js/dropdowns.js]: Same thing as above, let's scope to just look within `this.$panel`.

File: js/dropdowns.js [hunk 3]
```diff
@@ -106,14 +113,69 @@ Dropdown.prototype.handleCheck = function(e) {
...
+
+  $button.toggleClass('is-checked');
+};
+
+Dropdown.prototype.handleCheckboxRemoval = function($input) {
+  var $label = $input.parent().find('label');
+  var $button = $('button[data-label="' + $input.attr('id') +'"]');
```
Comment (noahmanger) [inline js/dropdowns.js]: Another place to scope within `this.$panel`

File: js/filters.js [hunk 4]
```diff
@@ -137,6 +137,11 @@ Filter.prototype.handleChange = function(e) {
 
     if (loadedOnce) {
       $label.addClass('is-loading');
+
+      // dropdown loading status
+      if ($input.parent().hasClass('dropdown__item')) {
+        $('button[data-name="' + $input.attr('name') + '"]').addClass('is-loading');
```
Comment (noahmanger) [inline js/filters.js]: Should we also scope this to `this.$body.find('button...)` as well?

File: scss/components/_dropdowns.scss [hunk 5]
```diff
@@ -33,6 +33,25 @@
...
+      padding-left: u(1.5rem);
+      right: u(1rem);
+    }
+  }
+
+  &.is-successful {
+    &::after {
+      @include u-icon-button($check);
+      background-position: 1.25rem 50%;
+      background-size: 18px;
```
Comment (noahmanger) [inline scss/components/_dropdowns.scss]: Should use `rem` throughout, (as `u(...rem)`)

File: scss/components/_dropdowns.scss [hunk 6]
```diff
@@ -99,7 +118,10 @@
 
       &.is-checked {
         @include u-icon-bg($check, $green-light);
-        background-position: 98%;
+        background-color: $gray-light;
+        background-position: 96%;
```
Comment (noahmanger) [inline scss/components/_dropdowns.scss]: Let's use the new CSS shorthand I just learned! `background-position: right 1rem;` or whatever the offset from the right side is.

File: tests/dropdowns.js [hunk 7]
```diff
@@ -85,7 +85,7 @@ describe('dropdown', function() {
     var selectedItems = this.dropdown.$selected.find('.dropdown__item');
     var panelItems = this.dropdown.$panel.find('.dropdown__item');
     expect(selectedItems.length).to.equal(1);
-    expect(panelItems.length).to.equal(1);
+    expect(panelItems.length).to.equal(2);
```
Comment (noahmanger) [inline tests/dropdowns.js]: Should we add a test to specifically check that the button was added?

File: scss/components/_dropdowns.scss [hunk 8]
```diff
@@ -149,6 +178,17 @@
     max-width: 90%;
   }
 
+  .remove {
```
Comment (noahmanger) [inline scss/components/_dropdowns.scss]: How do you feel about a more specific name for this? Maybe `.dropdown__remove`?

**Comment by noahmanger** [comment]
Works great! I left some picky comments throughout, but by and large it's solid. 

One small thing I noticed is that the remove buttons don't show up in Firefox. Not a big deal, but worth looking into.

File: scss/components/_dropdowns.scss (hunk 6 above)
Comment (xtine) [inline scss/components/_dropdowns.scss]: Oh I didn't know that, that's so much easier, ha.

Comment (noahmanger) [inline scss/components/_dropdowns.scss]: Right?! I just learned it yesterday :)

File: tests/dropdowns.js (hunk 7 above)
Comment (xtine) [inline tests/dropdowns.js]: Just added a bunch of tests. :)

Comment (noahmanger) [inline tests/dropdowns.js]: Yay!

**Pull request merged**